discojs/src/validation/validator.ts

```diff
@@ -21,23 +21,22 @@ export class Validator {
     }
   }
 
-  private async getLabel (ys: tf.Tensor): Promise<Float32Array | Int32Array | Uint8Array> {
-    switch (ys.shape[1]) {
-      case 1:
-        return await ys.greaterEqual(tf.scalar(0.5)).data()
-      case 2:
-        return await ys.argMax(1).data()
-      default:
-        throw new Error(`unable to reduce tensor of shape: ${ys.shape.toString()}`)
+  private async getLabel(ys: tf.Tensor): Promise<Float32Array | Int32Array | Uint8Array> {
+    // Binary classification
+    if (ys.shape[1] == 1) {
+      return await ys.greaterEqual(tf.scalar(0.5)).data()
+    // Multi-class classification
+    } else {
+      return await ys.argMax(-1).data()
     }
+    // Multi-label classification is not supported
```

Comment:
consider throwing here

Reply:
Hmm I don't think there is a way to detect if the user is doing multi-label classification with the output logits only. I think it will be quite obvious when creating a new task that there is no setting allowing multiple labels for one datapoint.